docker/vue-app/src/components/dialog_content/IndicatorEditingDialog.js

```diff
@@ -0,0 +1,663 @@
+            if(typeof this[dataPropertyName] === 'boolean') {
+                this[dataPropertyName] = !this[dataPropertyName];
+            }
+        },
+        getFormParentIDs() {
+            return new Promise((resolve, reject)=> {
+                $.ajax({
+                    type: 'GET',
+                    url: `${this.APIroot}/form/_${this.currCategoryID}/flat`,
+                    success: (res)=> {
+                        for (let i in res) {
+                            res[i]['1'].name = XSSHelpers.stripAllTags(res[i]['1'].name);
+                        }
+                        resolve(res);
+                    },
+                    error: (err)=> reject(err)
+                });
+            });
+        },
+        preventSelectionIfFormatNone() {
+            if (this.format === '' && (this.required === true || this.is_sensitive === true)) {
+                this.required = false;
+                this.is_sensitive = false;
+                alert(`You can't mark a field as sensitive or required if the Input Format is "None".`);
+            }
+        },
+        onSave(){
+            //check for advanced text formatting for name field
+            const elTrumbow = document.querySelector('.trumbowyg-editor');
+            if(elTrumbow !== undefined && elTrumbow !== null){
+                this.name = elTrumbow.innerHTML;
+            }
+            
+            let indicatorEditingUpdates = [];
+            if (this.isEditingModal) { /*  CALLS FOR EDITTING AN EXISTING QUESTION */
+                const nameChanged = this.name !== this.ajaxIndicatorByID[this.currIndicatorID].name;
+                const descriptionChanged = this.description !== this.ajaxIndicatorByID[this.currIndicatorID].description;
+
+                const options = this.ajaxIndicatorByID[this.currIndicatorID]?.options ? 
+                                '\n' + this.ajaxIndicatorByID[this.currIndicatorID]?.options?.join('\n') : '';
+                const fullFormatChanged = this.fullFormatForPost !== this.ajaxIndicatorByID[this.currIndicatorID].format + options;
+
+                const defaultChanged = this.defaultValue !== this.ajaxIndicatorByID[this.currIndicatorID].default;
+                const requiredChanged = +this.required !== parseInt(this.ajaxIndicatorByID[this.currIndicatorID].required);
+                const sensitiveChanged = +this.is_sensitive !== parseInt(this.ajaxIndicatorByID[this.currIndicatorID].is_sensitive);
+                //const sortChanged = this.sort !== parseInt(this.ajaxIndicatorByID[this.currIndicatorID].sort);
+                const parentIDChanged = this.parentID !== this.ajaxIndicatorByID[this.currIndicatorID].parentID;
+                const shouldArchive = this.archived === true;
+                const shouldDelete = this.deleted === true;
+                console.log('CHANGES?: name,descr,fullFormat,default,required,sensitive,sort,parentID,archive,delete');

```

Comment:
What's the point of these console logs?  I know we're doing that with errors, but that's a stop-gap for the error class when we get it.  We shouldn't be sending regular stuff to console.

Reply:
commented these out.  They will be completely removed when this is live